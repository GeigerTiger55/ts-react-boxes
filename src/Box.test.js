import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", function () {
  render(<Box
    id='123abc'
    height={3}
    width={2}
    backgroundColor='blue'
    remove={() => { }}
  />);
});

it("matches snapshot", function () {
  const { container } = render(<Box
    id='123abc'
    height={3}
    width={2}
    backgroundColor='blue'
    remove={() => { }}
  />);
  expect(container).toMatchSnapshot();
});

it("makes a div with matching properties", function () {
  const { container, debug } = render(<Box
    id='123abc'
    height={3}
    width={2}
    backgroundColor='blue'
    remove={() => { }}
  />);
  debug(container.querySelector('div[class="Box-box"]'));
  expect(
    container.querySelector('div[class="Box-box"]')
  ).toBeInTheDocument();
});
