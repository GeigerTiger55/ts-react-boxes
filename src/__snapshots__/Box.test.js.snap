// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 3em; width: 2em; background-color: blue;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
